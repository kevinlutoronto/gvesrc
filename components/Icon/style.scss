@import '../../styles/base';

.wrapper {
  fill: $cGray50;
  display: inline-block;
  margin: 0 10px;
}
