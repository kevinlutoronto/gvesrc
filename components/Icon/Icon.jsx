// @flow
import React from 'react';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';
import style from './style.scss';

type Props = {|
  icon: string,
  color: string,
  data?: Data,
  size?: number,
|};

const Icon = ({ icon, color, data }: Props) => {
  return (
    <div
      className={style.wrapper}
      style={{ fill: color }}
      dangerouslySetInnerHTML={{ __html: icon }}
      {...getDataAttrs(data)}
    />
  );
};

Icon.defaultProps = {
  color: '#000',
  size: 16,
};

Icon.displayName = 'Plasma@Icon';

export default Icon;
