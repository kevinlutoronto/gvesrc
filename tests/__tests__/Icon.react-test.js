/* eslint-disable */

import React from 'react';
import Icon from '../../components/Icon/Icon';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Icon Component', () => {
  test('render', () => {
    const component = renderer.create(<Icon />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
