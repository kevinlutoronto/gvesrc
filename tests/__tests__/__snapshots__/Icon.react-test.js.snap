// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component render 1`] = `
<div
  className="wrapper"
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
  style={
    Object {
      "fill": "#000",
    }
  }
/>
`;
